### ISSUE TITLE
Automatically install wsl 18.04 (from powershell)

### ISSUE THREAD

**Issue by a-t-0** [a-t-0/Taskwarrior-installation#104]
Automatically install the wsl from the exe. It might be easiest to use the following information to do this:

According to this site: https://docs.microsoft.com/en-us/windows/wsl/install-manual you can install the WSL ubuntu 18.04 automatically in powershell with commands:

0. Open powershell (in administrative mode).
1. Enable the permission to download stuff in powershell:
1.1 Source: https://stackoverflow.com/questions/41618766/powershell-invoke-webrequest-fails-with-ssl-tls-secure-channel
1.2 What the command does: _Use SecurityProtocolType.Ssl2 if needed for compatibility reasons_
```
[Net.ServicePointManager]::SecurityProtocol = [Net.SecurityProtocolType]::Tls12
```
2. Followed by the command to actually download wsl 18.04 (it downloads it to the current directory of powershell terminal):
```
Invoke-WebRequest -Uri https://aka.ms/wsl-ubuntu-1804 -OutFile Ubuntu.appx -UseBasicParsing
```
 3. The next command can be used to install the app using powershell:
```
Add-AppxPackage .\app_name.appx
```
3.1 Where you replace `app_name` with the file name that is downloaded, if you used the above commands its:
```
Add-AppxPackage .\Ubuntu.appx
```

4. First installing the wsl ubuntu via manual `.appx` file yields permission error.
4.1 The following command is suggested to enable hidden administrator mode:
```
net user administrator/active: yes
```
4.2 Also uninstall all apps from powershell admin:
Source: https://answers.microsoft.com/en-us/windows/forum/all/getting-windows-cannot-access-the-specified-device/9b534c5a-ec5a-41f3-ab30-e7c295ebdbf0
```
WSreset.exe
Get-AppxPackage -allusers Microsoft.WindowsStore | Remove-AppxPackage
```
4.3 re-register all apps: 
``` 
Get-AppXPackage -AllUsers | Foreach {Add-AppxPackage -DisableDevelopmentMode -Register "$($_.InstallLocation)\AppXManifest.xml"}
```
Or you could download the 18.04 arm version from:
https://docs.microsoft.com/en-us/windows/wsl/install-manual
with link:
https://aka.ms/wsl-ubuntu-1804-arm

and open the exe.

**Comment by a-t-0** [comment]
Note, without a valid microsoft-account sign-in the following commands succesfully installed WSL Ubuntu 18.04:

```
Enable-WindowsOptionalFeature -Online -FeatureName Microsoft-Windows-Subsystem-Linux
[Net.ServicePointManager]::SecurityProtocol = [Net.SecurityProtocolType]::Tls12
Invoke-WebRequest -Uri https://aka.ms/wsl-ubuntu-1804 -OutFile Ubuntu1804.appx -UseBasicParsing
Add-AppxPackage .\Ubuntu1804.appx
```

**Comment by a-t-0** [comment]
Solved in commit #2e8ee54. (Technically a bit earlier but this included a fix to also check the necessary files for the `AutoInstallTaskwarrior.jar` and skip of the check inside that `.jar` file which could cause the `.jar` to wait indefinitely if the files were not present at the start.).
